src/classes/vehicle.py

```diff
@@ -29,15 +29,16 @@ def __init__(self,
                  image_source: str,
                  ):
         self.id = id
-        self.position = position
-        self.velocity = velocity
-        self.acceleration = acceleration
-        self.direction = direction
+        self.position = position.copy()
+        self.velocity = velocity.copy()
+        self.acceleration = acceleration.copy()
+        self.direction = direction.copy()
         self.width = width
         self.length = length
         self.pivot_distance = pivot_distance
         self.intent = intent
-        self.image = pygame.image.load(image_source)
```

Comment:
what's the reason for keeping image source?

Reply:
For the reload, deep_copy doesn't support some pygame image object thing, so I had to resort to keeping the image source for loading the image.